src/vector/faer_serial.rs

```diff
@@ -0,0 +1,191 @@
+use std::ops::{Div, Mul, MulAssign};
+
+use faer::{unzipped, zipped, Col, ColMut, ColRef};
+
+use crate::{scalar::Scalar, scalar::Scale, IndexType};
+
+use super::{Vector, VectorCommon, VectorIndex, VectorView, VectorViewMut};
+impl<'a, T: Scalar> Mul<Scale<T>> for ColRef<'a, f64> {
+    type Output = Col<f64>;
+    fn mul(self, rhs: Scale<T>) -> Self::Output {
+        self * faer::scale(rhs.value().into())
+    }
+}
+
+impl<'a, T: Scalar> Mul<Scale<T>> for ColMut<'a, f64> {
+    type Output = Col<f64>;
+    fn mul(self, rhs: Scale<T>) -> Self::Output {
+        self * faer::scale(rhs.value().into())
+    }
+}
+
+impl<'a, T: Scalar> Mul<Scale<T>> for Col<f64> {
+    type Output = Col<f64>;
+    fn mul(self, rhs: Scale<T>) -> Self::Output {
+        self * faer::scale(rhs.value().into())
+    }
+}
+impl<'a, T: Scalar> MulAssign<Scale<T>> for ColMut<'a, f64> {
+    fn mul_assign(&mut self, rhs: Scale<T>) {
+        let scale = faer::scale(rhs.value().into());
```

Comment:
same with the mul_assigns. Do the two bodies need to be different?

Reply:
They have to, since faer implements the multiplication for some types and not for others